packages/ui/src/fields/Relationship/index.tsx

```diff
@@ -122,706 +61,149 @@ const RelationshipFieldComponent: RelationshipFieldClientComponent = (props) =>
-                        setValue({
-                          relationTo: selected.relationTo,
-                          value: selected.value,
-                        })
-                      } else if (!Array.isArray(selected)) {
-                        setValue(selected.value)
-                      }
-                    }
-                  : undefined
-              }
-              onInputChange={(newSearch) => handleInputChange(newSearch, value)}
-              onMenuClose={() => {
-                setMenuIsOpen(false)
-              }}
-              onMenuOpen={() => {
-                if (appearance === 'drawer') {
-                  openListDrawer()
-                } else if (appearance === 'select') {
-                  setMenuIsOpen(true)
-                  if (!hasLoadedFirstPageRef.current) {
-                    setIsLoading(true)
-                    void getResults({
-                      filterOptions,
-                      lastLoadedPage: {},
-                      onSuccess: () => {
-                        hasLoadedFirstPageRef.current = true
-                        setIsLoading(false)
-                      },
-                      value: initialValue,
-                    })
-                  }
-                }
-              }}
-              onMenuScrollToBottom={() => {
-                void getResults({
-                  filterOptions,
-                  lastFullyLoadedRelation,
-                  lastLoadedPage,
-                  search,
-                  sort: false,
-                  value: initialValue,
-                })
-              }}
-              options={options}
-              placeholder={placeholder}
-              showError={showError}
-              value={valueToRender ?? null}
-            />
-            {!(readOnly || disabled) && allowCreate && (
-              <AddNewRelation
-                hasMany={hasMany}
-                path={path}
-                relationTo={relationTo}
-                setValue={setValue}
-                value={value}
-              />
-            )}
-          </div>
-        )}
-        {errorLoading && <div className={`${baseClass}__error-loading`}>{errorLoading}</div>}
-        {AfterInput}
-        <RenderCustomComponent
-          CustomComponent={Description}
-          Fallback={<FieldDescription description={description} path={path} />}
-        />
-      </div>
-      {currentlyOpenRelationship.collectionSlug && currentlyOpenRelationship.hasReadPermission && (
-        <DocumentDrawer onDelete={onDelete} onDuplicate={onDuplicate} onSave={onSave} />
-      )}
-      {appearance === 'drawer' && !readOnly && (
-        <ListDrawer allowCreate={allowCreate} enableRowSelections={false} onSelect={onListSelect} />
-      )}
-    </div>
+    <RelationshipInput
+      {...sharedProps}
+      {...(hasMany === true
+        ? {
+            hasMany: true,
+            initialValue: memoizedInitialValue as ValueWithRelation[],
```

Comment:
The `initialValue` and `memoizedValue` doesn't need to be contained in this conditional spread. No need to assert the type here.